### PR TITLE
 Rails Group Project.

### PR DESCRIPTION
**Pull request by piero-vic**
## Purpose
This PR encompasses the entire work required for the Ruby on Rails group project.

## Features That we made here as a team 💪 💯 🍄 
- Followed the provided ERD, wireframes, and guidelines 
- Built Login page and registration page with Devise.
- Food list:
  - Show a list of food as in the wireframe.
  - Lead to the form that allows users to add new food.
- Recipes list:
  - Show a list of recipes created by the logged-in user as in the wireframe.
  - Lead to recipe details.
  -  When the user is the owner of the recipe, allow the user to delete it.
- Recipes details:
  - When the user is the owner of the recipe, allow the user to update it.
-  Public recipe list:
   - Show a list of all public recipes ordered by newest as in the wireframe.
   - Lead to recipe details.
- Add `README.md` file.
- No linters error. ✨💯🚀

## IMPORTANT:

Please, seed the test database before running the test.

```
rails db:seed RAILS_ENV=test
rspec spec
```

### CODE REVIEW

**Review by bimbolabuari — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Set up the repository on GitHub and use Gitflow.
- Uses Devise for authentication

## Required Changes ♻️
- Kindly write Unit and Integration tests for model, request, route as required

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/food_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    end
+
+    it 'I can see the button add food' do
+      expect(page).to have_content('Add food')
+    end
+
+    # it 'When I click on add foos, I am redirected to that form add food page.' do
+    #   click_link 'Add food'
+    #   expect(page.current_path).to have_content("foods/#{food.id}")
+    # end
```
Comment (TedLivist) [inline spec/views/food_index_spec.rb]: - [ ] I would recommend you remove commented code to make your tests, look neat.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @piero-vic and @Reem-lab,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
- [ ] Most of your integration tests are failing. Kindly fix them.

  failing-tests

  <hr /> 

  ### OPINION
  This is probably because, in some of the related integration tests, you are visiting the login route and inputting a user that does not exist. Maybe you need to create a user in each test file before logging them in and visiting other relevant routes

- [ ] _Check the comments under the review._

## Optional suggestions

_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/food_index_spec.rb (hunk 1 above)
Comment (Reem-lab) [reply spec/views/food_index_spec.rb]: Hi @TedLivist 
Thanks for reviewing it
the tests run correctly but you have to run the migration seeds before it !!!
I write it in the pull request description you can check it, please!!
`rails db:seed RAILS_ENV=test`

**Comment by piero-vic** [comment]
Hi @TedLivist. 
There are instructions on the README to seed the test database running `rails db:seed RAILS_ENV=test` before running tests. I think you missed that. I will leave this comment as a note for the next reviewer. Thanks for the feedback.

File: spec/views/food_index_spec.rb (hunk 1 above)
Comment (TedLivist) [reply spec/views/food_index_spec.rb]: Oh yeah @Reem-lab you are right. I missed the instructions due to oversight. Your tests are passing.

Comment (Whoistolu) [reply spec/views/food_index_spec.rb]: Well resolved 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi Team
@piero-vic and @Reem-lab 

Great job implementing the recommended changes 👍🏽 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**